### DISCUSSION TITLE
Override fieldTypes for new field types

### DISCUSSION THREAD

**Discussion by icedcola** [puckeditor/puck#702] [Q&A] [answered]
Hello. I'm having some trouble defining new field types with the overrides. Just wondering if I am doing something silly, or if the overrides aren't meant for this.

I have an override like so (just to test):
```ts
const overrides = {
    fieldTypes: {
      slider: ( { name, onChange, value }) => (<div>Slide!</div>),
    },
```

and using it in a component:
```ts
fields: {
        Sliderbar: {label: "Test", type: "slider"}
        },
```

gives an `Uncaught TypeError: defaultFields[field.type] is not a function at AutoFieldInternal` error.

The Docs mention "You can specify a custom render method for each known [field type](https://puckeditor.com/docs/api-reference/fields), **or introduce completely new ones.**"

I replaced with a `text` or `number` override and that seems to work. Assuming it is only for overriding default field types, but hopefully I'm just messing something up :) Thank you for any help.

Best regards,
Matt

**Comment by chrisvxd** [comment, marked as answer]
Hey @icedcola! Sorry for the delay, but I suspect this might be a TS bug. Going to create an issue.

**Reply by icedcola** [reply]
No worries at all. Ah, cool - at least it wasn't me :) Thank you for your time.
